src/main/java/com/sps/Servlets/NewRecipeServlet.java

```diff
@@ -134,13 +136,13 @@ private void addToSearchStrings(Collection<String> searchStrings, String stringT
   private Recipe entityToRecipe(Entity recipeEntity) {
     String name = (String) recipeEntity.getProperty("name");
     String description = (String) recipeEntity.getProperty("description");
-    LinkedList<String> tags = (LinkedList<String>) (LinkedList<?>) getDataAsList(recipeEntity.getProperty("tags"), TAG);
-    LinkedList<String> ingredients = (LinkedList<String>) (LinkedList<?>) getDataAsList(recipeEntity.getProperty("ingredients"), INGREDIENT);
+    LinkedHashSet<String> tags = new LinkedHashSet<>((LinkedList<String>) (LinkedList<?>) getDataAsList(recipeEntity.getProperty("tags"), TAG));
```

Comment:
y the change :0

Reply:
I kept going back and forth between sets and lists. Sets make sure there are never any duplicates but don't preserve order, while lists preserve order but don't prevent duplicates. Then I discovered LinkedHashSets, which give me the best of both worlds so order is maintained AND there can't be duplicates.